src/OrchardCore/OrchardCore.ContentManagement/FileContentDefinitionStore.cs

```diff
@@ -65,6 +65,12 @@ public Task SaveContentDefinitionAsync(ContentDefinitionRecord contentDefinition
         {
             lock (this)
             {
+                var directoryPath = Path.GetDirectoryName(Filename);
```

Comment:
Nice catch!! but I'm not sure how this work in Windows or MacOS? @rserj could please test in on other OS

Thanks

Reply:
@hishamco thank you for your review. 
In the image above I reproduced issue on Linux, I commented out the fix and was able to reproduce issue on Windows, please see image below
![image](https://user-images.githubusercontent.com/13991195/87701105-422e6480-c74c-11ea-857b-693818f05fc9.png)
I tested fix on Windows and Linux as well, it works fine, unfortunately, I can not test on MasOS, but following to API documentation
https://docs.microsoft.com/en-us/dotnet/api/system.io.file.createtext?view=netcore-3.1
DirectoryNotFoundException is expected behavior, 
>DirectoryNotFoundException
The specified path is invalid (for example, it is on an unmapped drive).

It only fails if you try to install Agency recipe with some of the external Database such as MySql